### PR TITLE
[FE-151] [FE-152] All levels section, update progress bar

### PR DESCRIPTION
**Pull request by AnCichocka**
https://oops2000.atlassian.net/browse/SCRUM-151
https://oops2000.atlassian.net/browse/SCRUM-152

- add levels section
- update progress bar
- add Level type
- add index file

### CODE REVIEW

File: frontend/src/hooks/StudentProfile/useStudentProfileData/useAnimalData.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return undefined;
+  }
+  return {
+    name: data.levelName,
+    ordinalNumber: data.ordinalNumber,
+    realLevelNumber: data.ordinalNumber + 1,
+    imageId: data.imageFile?.fileId ?? undefined,
+    // TODO should be int of float
+    minimumPoints: parseInt(data.minimumPoints),
+    maximumPoints: parseInt(data.maximumPoints),
```
Comment (AnCichocka) [inline frontend/src/hooks/StudentProfile/useStudentProfileData/useAnimalData.tsx:26]: Should it be `parseInt` of `parseFloat`?

File: frontend/src/components/StudentProfile/cards/AnimalCard/LevelsSection.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  if (loading) return <div>loading...</div>;
+  if (error) return <div>ERROR: {error.message}</div>;
+
+  return (
+    <div style={styles.container}>
+      <div style={styles.title}>All levels</div>
+      {levels.length > 0 ? (
+        <div style={styles.levelsContainer}>
+          {levels?.map((level) => (
+            <AnimalWithTooltip
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/cards/AnimalCard/LevelsSection.tsx:23]: TODO correct name, I missed that

File: frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const leftLevel = prevLevel ?? currLevel;
+  const rightLevel = nextLevel ?? currLevel;
+
+  return (
+    <div style={styles.container}>
+      <ProgressBar
+        points={totalPoints - currLevel.minimumPoints}
+        bounds={{
+          lower: 0,
+          upper: currLevel.maximumPoints - currLevel.minimumPoints,
```
Comment (AnCichocka) [inline frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx:28]: @Szyntos please check if it's correct, I mean when min = 10 and max = 20 should I display interval 10-20, 11-20, 10-19 or 10-20?

File: frontend/src/components/images/AnimalWithTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  disabled,
+}: AnimalWithTooltipProps) => {
+  return (
+    <TooltipWrapper
+      tooltipContent={
+        <div style={styles.container}>
+          <div style={styles.title}>{level.name}</div>
+          <div>lvl. {level.realLevelNumber}</div>
+          <div>
+            from {level.minimumPoints} to {level.maximumPoints} points
```
Comment (AnCichocka) [inline frontend/src/components/images/AnimalWithTooltip.tsx:24]: @Szyntos here as well

File: frontend/src/hooks/StudentProfile/useLevelsData.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { useEditionSelection } from "../common/useEditionSelection";
+import { useLevelsQuery } from "../../graphql/levels.graphql.types";
+import { Level } from "./types";
+
+export const useLevelsData = () => {
+  const { selectedEdition } = useEditionSelection();
+  const editionId = selectedEdition?.editionId;
+
+  const { data, error, loading } = useLevelsQuery({
+    variables: { editionId: editionId },
```
Comment (Szyntos) [inline frontend/src/hooks/StudentProfile/useLevelsData.tsx:10]: image
userId = 15

useLevelsData queries for editionId=2, where in fact this user is in editionId=1

File: frontend/src/hooks/StudentProfile/useStudentProfileData/useAnimalData.tsx (hunk 1 above)
Comment (Szyntos) [reply frontend/src/hooks/StudentProfile/useStudentProfileData/useAnimalData.tsx:25]: This is a question whether we should allow floating point level range. On Backend we're allowing it, so i guess we should change it to float

File: frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx (hunk 3 above)
Comment (Szyntos) [reply frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx:28]: IMO the interval should be displayed as

```[10, 20), [20, 30)```

the lower bound is inclusive, the upper is exclusive

**Review by Szyntos — CHANGES_REQUESTED** [review]
One error with userId=15

File: frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx (hunk 3 above)
Comment (AnCichocka) [reply frontend/src/components/bars/LevelProgressBar/LevelProgressBar.tsx:28]: So i should do 10-19? or leave 20 to display?

File: frontend/src/hooks/StudentProfile/useLevelsData.tsx (hunk 5 above)
Comment (AnCichocka) [reply frontend/src/hooks/StudentProfile/useLevelsData.tsx:10]: > image userId = 15
> 
> useLevelsData queries for editionId=2, where in fact this user is in editionId=1

maybe that's the case xd

`query Levels($editionId: bigint) {
  levels(where: { editionId: { _eq: 2 } }, orderBy: { ordinalNumber: ASC }) {
    grade
    imageFileId
    levelId
    maximumPoints
    minimumPoints
    name
    ordinalNumber
    label
    highest
    editionId
  }
}
`

**Comment by AnCichocka** [comment]
> One error with userId=15

now it works 💪

File: frontend/src/graphql/levels.graphql [hunk 6]
```diff
@@ -0,0 +1,17 @@
+query Levels($editionId: bigint) {
```
Comment (Szyntos) [inline frontend/src/graphql/levels.graphql:1]: 1. It should be Int!

```graphql
query Levels($editionId: Int!) {
  levels(
    where: { editionId: { _eq: editionId } }
    orderBy: { ordinalNumber: ASC }
  ) {
    grade
    imageFileId
    levelId
    maximumPoints
    minimumPoints
    name
    ordinalNumber
    label
    highest
    editionId
  }
}
```

2. I get this error:

image


```json
{
    "errors": [
        {
            "message": "expecting a value for non-nullable variable: \"editionId\"",
            "extensions": {
                "path": "$",
                "code": "validation-failed"
            }
        }
    ]
}
```

You're not sending variables somehow

Comment (Szyntos) [reply frontend/src/graphql/levels.graphql:1]: Even without the change to Int! there is the same error:

image

image

**Review by Szyntos — CHANGES_REQUESTED** [review]

File: frontend/src/graphql/levels.graphql (hunk 6 above)
Comment (AnCichocka) [reply frontend/src/graphql/levels.graphql:1]: sus

**Review by Szyntos — APPROVED** [review]
WORKSSSSS

**Pull request merged**